### PR TITLE
Refactor/extend mockgraph by satellite

### PR DESCRIPTION
**Pull request by romanatarango**
### Scope & Purpose

Extended `Testgraphs/allLocalGraph` and `Testgraphs/allRemoteGraph`:
1. Introduced satellite vertices to `MockSmartGraph`s.
2. Implemented tests from the ticket `GORDO-1390`.

- [ ] :hankey: Bugfix
- [x] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [x] Tests
  - [ ] **Regression tests**
  - [x] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*
  - [ ] Backport for 3.7: *(Please link PR)*

#### Related Information

- [ ] Docs PR: 
- [x] Enterprise PR: https://github.com/arangodb/enterprise/pull/929
- [ ] GitHub issue / Jira ticket: GORDO-1390 (https://arangodb.atlassian.net/browse/GORDO-1390?atlOrigin=eyJpIjoiYTU4Y2RlNDVjNDcwNDgyOTkxNTFiNTZmYTRkZmRkMDMiLCJwIjoiaiJ9)
- [ ] Design document:

### CODE REVIEW

File: arangod/VocBase/LogicalCollection.cpp [hunk 1]
```diff
@@ -387,34 +393,47 @@ uint64_t LogicalCollection::numberDocuments(transaction::Methods* trx,
...
+}
 
-TRI_vocbase_col_status_e LogicalCollection::status() const { return _status; }
+TRI_vocbase_col_status_e LogicalCollection::status() const {
+  TRI_ASSERT(_sharding != nullptr);
+  return _status;
+}
 
 TRI_vocbase_col_status_e LogicalCollection::getStatusLocked() {
+  TRI_ASSERT(_sharding != nullptr);
```
Comment (hkernbach) [inline arangod/VocBase/LogicalCollection.cpp:423]: Are all of those new asserts really required here? Seems to be added to every method:
```
  TRI_ASSERT(_sharding != nullptr);
```

File: arangod/Graph/Providers/SingleServerProvider.h [hunk 2]
```diff
@@ -110,6 +110,13 @@ class SingleServerProvider {
 
   void prepareContext(aql::InputAqlItemRow input);
   void unPrepareContext();
+  /**
+ * Return true if the vertex whose id is stored in the class (in _vertex) has
+ * its data on this DB-server.
+ * @param trx
+ * @return
```
[the remark this answers is not in the document]
Comment (hkernbach) [reply arangod/Graph/Providers/SingleServerProvider.h:117]: trx not used here, return not defined -> therefore can just be deleted OR modified properly.

**Review by hkernbach — CHANGES_REQUESTED** [review]
Just minor things. One comment is not finalized / complete. Then I am unsure about all the added asserts to LogicalCollection.

File: arangod/VocBase/LogicalCollection.cpp (hunk 1 above)
Comment (romanatarango) [reply arangod/VocBase/LogicalCollection.cpp:423]: Michael's decision.

Comment (mchacki) [reply arangod/VocBase/LogicalCollection.cpp:423]: Uh I think we had a misunderstanding here.
is the _sharding is accessed within the method it shall be asserted.
If not, it shall not be asserted. (it is not required.)

File: arangod/VocBase/LogicalCollection.cpp [hunk 3]
```diff
@@ -611,6 +619,7 @@ void LogicalCollection::toVelocyPackForClusterInventory(VPackBuilder& result,
                                                         bool useSystem,
                                                         bool isReady,
                                                         bool allInSync) const {
+  TRI_ASSERT(_sharding != nullptr);
```
Comment (hkernbach) [inline arangod/VocBase/LogicalCollection.cpp:622]: Still here.

**Review by hkernbach — APPROVED** [review]

**Comment by hkernbach** [comment]
LGTM. There are a few merge conflicts that needs to be solved (Community + EE). After those are fixed, we need to start another jenkins run.

**Comment by mchacki** [comment]
Matrix PR failed on an unrelated issue.
This PR is considered green.

**Pull request merged**